pkg/scheduler/framework/interface.go

```diff
@@ -177,6 +177,9 @@ func (s *Status) Message() string {
 	if s == nil {
 		return ""
 	}
+	if s.err != nil && s.code != Error {
+		return strings.Join(append(s.reasons, s.err.Error()), ", ")
+	}
 	return strings.Join(s.reasons, ", ")
 }
```

Comment:
Also make sure that `AsStatus(err error)` doesn't duplicate the error message.

Reply:
I'm doing this by checking s.code in `s.Message()`, I think we have to save both the error msg and err in `AsStatus` for `s.Reasons()` to return correctly.